src/goto-programs/goto_convert_functions.cpp

```diff
@@ -163,13 +163,36 @@ void goto_convert_functionst::convert_function(
     symbol.is_compiled()) /* goto_inline may have removed the body */
     return;
 
+  const size_t ERROR_MSG_LEN = 512;
+  char parameter_error_message[ERROR_MSG_LEN];
   // we have a body, make sure all parameter names are valid
   for(const auto &p : f.parameter_identifiers)
   {
-    DATA_INVARIANT(!p.empty(), "parameter identifier should not be empty");
+    if (p.empty())
+    {
+      snprintf(parameter_error_message,
+               ERROR_MSG_LEN - 1,
+               "parameter identifier should not be empty\n"
+               "function: %s\n"
+               "parameter: %s",
+               identifier.c_str(),
+               p.c_str());
+    }
+    DATA_INVARIANT(!p.empty(), parameter_error_message);
+
+    if (!symbol_table.has_symbol(p))
+    {
+      snprintf(parameter_error_message,
+               ERROR_MSG_LEN - 1,
+               "parameter identifier must be a known symbol\n"
+               "function: %s\n"
+               "parameter: %s",
+               identifier.c_str(),
+               p.c_str());
+    }
     DATA_INVARIANT(
       symbol_table.has_symbol(p),
-      "parameter identifier must be a known symbol");
+      parameter_error_message);
```

Comment:
I'm not sure that defining and constructing all of this outside the invariant is the right choice. There is also `DATA_INVARIANT_WITH_DIAGNOSTICS` that handles other arguments beyond the message.

Reply:
Yes; what @TGWDB says.